### PR TITLE
Sorr task601 followup fix

### LINKED ISSUE #601: Marketing: Build a mailing list for AI VCs from Tra
https://docs.google.com/document/d/1z-svt4K6dSy5ncuVZzrtL9ZYuBmknCp8UfGfxljaBNQ/edit#heading=h.3peqjdp00fga
We would like to get the list of VCs investing in certain areas
We have a license of tracxn, we create some queries, scroll to the bottom to get all the entries and then we save it into files 
The idea is to read the saved html files and scrape them with Beautiful soup to generate CSV files / Google sheets with the fields above
After this we can use LinkedIn to extract the people working there

### PR DESCRIPTION
**Pull request by why2000**
#601 
FYI @samarth9008 @gpsaggese 
This is a follow up PR for #603 . 
* Simplified the workflow. Now the code can be applied directly on `.mht` or `.mhtml` files. No need to manual convert to html.
* Restructured the code into two libraries, and the notebooks using them. Both moved to `marketing` folder to be consistent with #606 .
* One thing to be marked up (also put in the comment): We won't be able to get any info if we download the webpage directly into an `html`. An embedded safekeeper code will erase everything before it is downloaded as an html. Which means `.mht` file format is a must-have when downloading it. Previously due to BeautifulSoup compatibility, we'll need to manually convert `mht` to `html` through a browser, now by applying the `quoted-printable` format decoder, it is no longer needed.

### CODE REVIEW

File: marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# %% run_control={"marked": true}
+import marketing.tra.extract_VCs_from_Tra_search_mhtml as mtevftsmh
+
+# %% [markdown]
+# # Workflow before using this library:
+#
+# 1. Go to a Tra VCs search result page
+# 2. Use the browser's `Download` or `Save As` button to download the webpage as a `Web page, single file`. Or use any other download method that can fulfill the requirement in step 3.
+# 3. If you see the downloaded file format is `.mht` or `.mhtml`, you can process forward. Otherwise you won't be able to bypass the check layer from the website.
+# 4. Use the `.mhtml` file path as the parameter to `get_VCs_from_mhtml` method, it will return a dataframe. Save it to whatever format preferred.
```
Comment (samarth9008) [inline marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py:24]: Lets follow the coding convention of 80 chars per line.

File: marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# %% [markdown]
+# # Sample usage of the function.
+
+# %% run_control={"marked": true}
+# Source data file path.
+vc_mhtml_path = "../data/Investors _ VC _ Tra.mhtml"
```
Comment (samarth9008) [inline marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py:31]: Remove the extra space, also from the filename in the location

File: marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# Destination result file path.
+vc_csv_save_path = "../result_csv/Investors _ VC _ Tra.csv"
```
Comment (samarth9008) [inline marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py:33]: Same as above

File: marketing/tra/extract_people_from_Tra_company_mhtml.py [hunk 4]
```diff
@@ -62,34 +51,20 @@ def get_employee_contents(soup: BeautifulSoup) -> List[List[str]]:
...
+
+    :param mhtml_file_path: The path of the company's people page as an mhtml file
     :return: A pandas.DataFrame containing the people's name and LinkedIn profile link
     """
-    with open(html_file_path, encoding="utf-8") as employee_fp:
-        soup = BeautifulSoup(employee_fp)
+    with open(mhtml_file_path, "rb") as employee_fp:
+        soup = BeautifulSoup(
+            quopri.decodestring(employee_fp.read()), features="lxml"
+        )
```
Comment (samarth9008) [inline marketing/tra/extract_people_from_Tra_company_mhtml.py:64]: Can you explain with comment what it is doing? Same for the other file.

File: marketing/tra/extract_people_from_Tra_company_mhtml.py [hunk 5]
```diff
@@ -26,7 +15,7 @@
 def _get_employee_row_content(content_soup: BeautifulSoup) -> List[str]:
     """
     Get one row from the table in an HTML page.
-    
+
     :param content_soup: A div soup containing the people table
     :return: A list containing one row from the table.
```
Comment (samarth9008) [inline marketing/tra/extract_people_from_Tra_company_mhtml.py:20]: Not dot in the end.

File: marketing/tra/extract_VCs_from_Tra_search_mhtml.py [hunk 6]
```diff
@@ -200,7 +191,7 @@ def _get_VC_row_content(content_soup: BeautifulSoup) -> List[str]:
 def get_VC_contents(soup: BeautifulSoup) -> List[List[str]]:
     """
     Extract the table content from a VC search result page.
-    
+
     :param soup: The BeautifulSoup instance of the VC search result page soup
     :return: A 2D list containing the table content.
```
Comment (samarth9008) [inline marketing/tra/extract_VCs_from_Tra_search_mhtml.py:196]: Not dot in the end.

File: marketing/notebooks/SorrTask601_Extract_people_from_Tra_company_html.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+# %%
+import marketing.tra.extract_people_from_Tra_company_mhtml as mtepftcmh
+
+# %% [markdown]
+# # Workflow before using this library:
+#
+# 1. Go to a Tra company's people page ("\<company_page\>/people/currentteam")
+# 2. Use the browser's `Download` or `Save As` button to download the webpage as a `Web page, single file`. Or use any other download method that can fulfill the requirement in step 3.
+# 3. If you see the downloaded file format is `.mht` or `.mhtml`, you can process forward. Otherwise you won't be able to bypass the check layer from the website.
+# 4. Use the `.mhtml` file path as the parameter to `get_employees_from_html` method, it will return a dataframe. Save it to whatever format preferred.
```
Comment (samarth9008) [inline marketing/notebooks/SorrTask601_Extract_people_from_Tra_company_html.py:24]: 80 chars per line.

File: marketing/notebooks/SorrTask601_Extract_people_from_Tra_company_html.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+# %% [markdown]
+# # Sample usage of the library.
+
+# %%
+# Source data file path.
+employee_mhtml_path = "../data/Sequoia Capital _ Tra.mhtml"
```
Comment (samarth9008) [inline marketing/notebooks/SorrTask601_Extract_people_from_Tra_company_html.py:31]: Remove space as mentioned above.

File: marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py (hunk 2 above)
Comment (why2000) [reply marketing/notebooks/SorrTask601_Extract_VCs_from_Tra_search_mhtml.py:31]: This space is from auto formatted jupyter notebook. I don't have the space in the notebook at all, so unable to remove it.

**Review by gpsaggese — COMMENTED** [review]
LGTM

**Review by samarth9008 — APPROVED** [review]
LGTM

**Comment by why2000** [comment]
Should we merge this? @samarth9008 @gpsaggese

**Pull request merged**